test/test-DOMFetcher.js

```diff
@@ -0,0 +1,66 @@
+"use strict";
+
+const test = require("sdk/test");
+const tabs = require("sdk/tabs");
+const httpd = require("./lib/httpd");
+const {before, after} = require("sdk/test/utils");
+const {DOMFetcher} = require("lib/DOMFetcher");
+const {Cu} = require("chrome");
+const {doGetFile} = require("./lib/utils");
+
+const {TextDecoder, OS} = Cu.import("resource://gre/modules/osfile.jsm", {});
+const decoder = new TextDecoder();
+
+const PORT = 8099;
+const url = `http://localhost:${PORT}/dummy.html`;
+const otherUrl = `http://localhost:${PORT}/other-dummy.html`;
+
+let srv;
+let dummyDOM;
+
+// Compare two strings ignoring all the blank spaces
+function contentCompare(stringA, stringB) {
+  return stringA.replace(/[\s]+/g, "") === stringB.replace(/[\s]+/g, "");
+}
+
+exports["test DOM fetching when tab closed"] = function(assert, done) {
+  let domFetcher = new DOMFetcher((message)=> {
+    assert.equal(message.data.type, "document-content", "the message received have the right type");
+    assert.ok(contentCompare(message.data.data.data, dummyDOM), "fetched DOM should be equal to the url content");
+    domFetcher.uninit();
+    done();
+  });
+  function onReady(tab) {
+    tab.close();
+  }
+  tabs.once("ready", onReady);
+  tabs.open({url});
+};
+
+exports["test DOM fetching when location change"] = function(assert, done) {
```

Comment:
if i read this test correct, this test will obtain the DOM data the first time, then close the tab.
It should instead expect:

1. to be called twice
2. to be called with the right data